src/java.base/share/classes/java/lang/ref/Cleaner.java

```diff
@@ -87,8 +87,10 @@
  * <pre>{@code
  * public class CleaningExample implements AutoCloseable {
  *        // A cleaner, preferably one shared within a library
- *        private static final Cleaner cleaner = <cleaner>;
+ *        private static final Cleaner cleaner = Cleaner.create();
  *
+ *        // Static state class, capturing information necessary for
+ *        // cleanup, but no reference to the instance being cleaned
```

Comment:
```suggestion
 *        // State class captures information necessary for cleanup.
 *        // It must hold no reference to the instance being cleaned
 *        // and therefore it is a static inner class in this CleaningExample.
```

Would this be clearer?

Reply:
This is what I suggested and makes it clear that *must hold no reference to the instance being cleaned*.  Maybe you didn't notice it's still there? 
 
```
 *        // State class captures information necessary for cleanup.
 *        // It must hold no reference to the instance being cleaned
 *        // and therefore it is a static inner class in this CleaningExample
 ```

Reply:
I like this suggestion - calling out that the static-ness of the State class is due to the requirement to not hold any references.

Reply:
> Maybe you didn't notice it's still there?

Yeah. Sorry, @mlchung. Good suggestion!